### ISSUE TITLE
set-constant.js doesn't interpret empty string value without quotation

### ISSUE THREAD

**Issue by Yuki2718** [uBlockOrigin/uBlock-issues#1528]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

In short `youjizz.com##+js(set-constant, config.ads.desktopPreroll, )` in AG Base doesn't work. Ofc `youjizz.com##+js(set, config.ads.desktopPreroll, '')` works.

<details>
<summary>Screenshot</summary>

![youjizz](https://user-images.githubusercontent.com/58900598/111058388-b71b8f00-84d1-11eb-8a89-3bd8fece3f71.png)

</details>

### A specific URL where the issue occurs

[NSFW] `https://www.youjizz.com/`

### Steps to Reproduce

1. Enable AdGuard Base filter
2. Visit the site
3. Check the property

### Expected behavior:

Not sure if it should be addressed by uBO or by AG's filter compiler.

### Actual behavior:

Described

### Your environment

* uBlock Origin version: 1.33.3rc1
* Browser Name and version: Firefox 86.0.1
* Operating System and version: Windows 10

**Comment by uBlock-user** [comment]
By-design implementation - https://github.com/uBlockOrigin/uAssets/commit/5051610f0e2374955a03c54be42bbbe9115f05c7#diff-900e11b92d7a794d808f94785527c81c42a603347d27d0a4460db4228bcc9c3cR2132

**Comment by Yuki2718** [comment]
Then how about setting empty string as default value of the third parameter if not explicitly set?

**Comment by gorhill** [comment]
There is no need for quotes in uBO -- if AdGuard wants to be compatible with uBO's filters, then it should not require quotes.

**Comment by gorhill** [comment]
Sorry, spoke too soon (morning coffee etc.).

It has been a while but my thinking must have been that a default value does not make sense, because it's too arbitrary which one it would be (could be `undefined`, `0`, `false`, etc.), and thus `set-constant` does require the parameter, unlike some other cases where the default makes sense.

**Comment by Yuki2718** [comment]
Okay. To be clear the original form of the filter is `youjizz.com#%#//scriptlet("set-constant", "config.ads.desktopPreroll", "")` but the quotation (single or double doesn't matter on AG) is stripped at the time of server compilation for uBO version of the list. I'll open an issue in https://github.com/AdguardTeam/FiltersCompiler

**Comment by Yuki2718** [comment]
Fixed by AG side.
